src/components/AppIndex.vue

```diff
@@ -39,34 +60,126 @@ export default {
+            })
+        },
+        // An array of keyword objects that are currently in use for filtering
+        selectedKeywords() {
+            return this.keywordsWithCount.filter((kw) => this.keywordFilter.includes(kw.name))
+        },
+        // An array of those keyword objects that are currently not in use for filtering
+        unselectedKeywords() {
+            return this.keywordsWithCount.filter((kw) => ! this.selectedKeywords.includes(kw))
+        },
+        // An array of all recipes that are part in all filtered keywords
+        recipesFilteredByKeywords() {
+            let $this = this
+            return this.recipes.filter(function (r) {
+                if($this.keywordFilter.length == 0) {
+                    // No filtering, keep all
+                    return true
+                }
+                
+                if(r.keywords === null){
+                    return false
+                }
+                
+                function keywordInRecipePresent(kw,r) {
+                    if(! r.keywords) {
+                        return false;
                     }
+                    
+                    let keywords = r.keywords.split(',')
+                    return keywords.includes(kw.name)
                 }
-                return filtered
+                
+                return $this.selectedKeywords.map((kw) => keywordInRecipePresent(kw, r)).reduce((l,r) => l && r)
+            })
+        },
+        // An array of the finally filtered recipes, that is both filtered for keywords as well as string-based name filtering
+        filteredRecipes () {
+            let ret = this.recipesFilteredByKeywords
+            let $this = this
+            
+            if(this.filters){
+                ret = ret.filter(function (r) {
+                    return r.name.toLowerCase().includes($this.filters.toLowerCase())
+                })
             }
+            
+            return ret
+        },
+        // An array of keywords that are yet unselected but some visible recipes are associated
+        selectableKeywords() {
+            if (this.unselectedKeywords.length === 0) {
+                return []
+            }
+            
+            let $this = this
+            return this.unselectedKeywords.filter(function (kw) {
+                return $this.filteredRecipes.map(function (r) {
+                    return r.keywords && r.keywords.split(',').includes(kw.name)
+                }).reduce((l,r) => l || r, false)
+            })
+        },
+        // An array of known keywords that are not associated with any visible recipe
+        unavailableKeywords() {
+            return this.unselectedKeywords.filter((kw) => ! this.selectableKeywords.includes(kw))
+        },
+        // An array of recipe objects of all recipes with links to the recipes and a property if the recipe is to be shown 
+        recipeObjects() {
+            let filtered = this.filteredRecipes
```

Comment:
if you pass `this` to `map` you don’t need to create a temporary variable, see [the map syntax ref](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/map)

Reply:
Yeah, that makes sense, I was worried that the array would change throughout the invocation. That would render the mapping broken. I will update as you suggested as it seems quite fast in my machine.

Reply:
Haven’t thought of that.. we should keep that in mind in case somebody experiences some issues